app/models/manageiq/providers/autosde/inventory/collector/storage_manager.rb

```diff
@@ -26,6 +25,43 @@ def storage_resources
     end
   end
 
+  def addresses
+    @addresses ||= @manager.autosde_client.StorageHostApi.storage_hosts_get.map do |addresses|
+      addresses_array = []
```

Comment:
`addresses_array`  here is redundant. Each address should go into its own hash. No need to group them into arrays.
Look how it's done in all other collectors.